src/codecs/framers/octet_counting.rs

```diff
@@ -0,0 +1,394 @@
+use crate::codecs::{BoxedFramer, BoxedFramingError, FramingConfig};
+use bytes::{Buf, Bytes, BytesMut};
+use serde::{Deserialize, Serialize};
+use std::io;
+use tokio_util::codec::{LinesCodec, LinesCodecError};
+
+/// Config used to build a `OctetCountingCodec`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct OctetCountingDecoderConfig {
+    max_length: Option<usize>,
+}
+
+#[typetag::serde(name = "octet_counting")]
+impl FramingConfig for OctetCountingDecoderConfig {
+    fn build(&self) -> crate::Result<BoxedFramer> {
+        Ok(Box::new(match self.max_length {
+            Some(max_length) => OctetCountingCodec::new_with_max_length(max_length),
+            None => OctetCountingCodec::new(),
+        }))
+    }
+}
+
+/// Codec using the `Octet Counting` format as specified in
+/// https://tools.ietf.org/html/rfc6587#section-3.4.1.
+#[derive(Clone, Debug)]
+pub struct OctetCountingCodec {
+    other: LinesCodec,
```

Comment:
This might be more clearly named as `fallback` or similar.

Also, is this behavior specified in the RFC or is it something we've implemented for compatibility? I don't mind it, but it's probably worth documenting somewhere.

Reply:
The implementation is copied 1:1 from here: https://github.com/timberio/vector/blob/b1c64a8af4b5e07d8b560fe4f486768a92e55260/src/sources/syslog.rs#L187-L393

I'd propose we defer this discussion and open another issue to improve it